sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1208,63 +1253,39 @@ class Analyzer(
         // Category 1:
         // BroadcastHint, Distinct, LeafNode, Repartition, and SubqueryAlias
         case p: BroadcastHint =>
-          p
         case p: Distinct =>
-          p
         case p: LeafNode =>
-          p
         case p: Repartition =>
-          p
         case p: SubqueryAlias =>
-          p
 
         // Category 2:
         // These operators can be anywhere in a correlated subquery.
         // so long as they do not host outer references in the operators.
         case p: Sort =>
           failOnOuterReference(p)
-          p
         case p: RepartitionByExpression =>
           failOnOuterReference(p)
-          p
 
         // Category 3:
         // Filter is one of the two operators allowed to host correlated expressions.
         // The other operator is Join. Filter can be anywhere in a correlated subquery.
         case f @ Filter(cond, child) =>
           // Find all predicates with an outer reference.
-          val (correlated, local) = splitConjunctivePredicates(cond).partition(containsOuter)
+          val (correlated, local) =
+            splitConjunctivePredicates(cond).partition(SubExprUtils.containsOuter)
```

Comment:
Let us `import org.apache.spark.sql.catalyst.expressions.SubExprUtils._` at the beginning of the rule of `ResolveSubquery`

Thus, no change is needed after the import.